packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -271,16 +300,24 @@ export class SerializedStateManager {
  */
 export async function getLatestSnapshotInfo(
 	mc: MonitoringContext,
-	storageAdapter: Pick<
-		IDocumentStorageService,
-		"getSnapshot" | "getSnapshotTree" | "getVersions" | "readBlob"
-	>,
+	storageAdapter: ISerializedStateManagerDocumentStorageService,
 	supportGetSnapshotApi: boolean,
 ): Promise<ISnapshotInfo | undefined> {
 	return PerformanceEvent.timedExecAsync(
 		mc.logger,
 		{ eventName: "GetLatestSnapshotInfo" },
 		async () => {
+			if (supportGetSnapshotApi && storageAdapter.loadedGroupIdSnapshots !== undefined) {
+				const snapshot = await storageAdapter.getSnapshot?.({
```

Comment:
Also, maybe we want to assert here that "getSnapshot" is not undefined and then remove "?" as then if below assert hits because of this somehow then we will get confused.

Reply:
Yep, I can do that